Android.mk

```diff
@@ -0,0 +1,308 @@
+    src/core/utils/heap.cpp                    \
+    src/core/utils/jam_detector.cpp            \
+    src/core/utils/missing_strlcpy.c           \
+    src/core/utils/missing_strlcat.c           \
+    src/core/utils/missing_strnlen.c           \
+    src/core/utils/slaac_address.cpp           \
+    third_party/mbedtls/repo/library/md.c                   \
+    third_party/mbedtls/repo/library/md_wrap.c              \
+    third_party/mbedtls/repo/library/memory_buffer_alloc.c  \
+    third_party/mbedtls/repo/library/platform.c             \
+    third_party/mbedtls/repo/library/sha256.c               \
+    third_party/mbedtls/repo/library/bignum.c               \
+    third_party/mbedtls/repo/library/ccm.c                  \
+    third_party/mbedtls/repo/library/cipher.c               \
+    third_party/mbedtls/repo/library/cipher_wrap.c          \
+    third_party/mbedtls/repo/library/cmac.c                 \
+    third_party/mbedtls/repo/library/ctr_drbg.c             \
+    third_party/mbedtls/repo/library/debug.c                \
+    third_party/mbedtls/repo/library/ecjpake.c              \
+    third_party/mbedtls/repo/library/ecp_curves.c           \
+    third_party/mbedtls/repo/library/entropy.c              \
+    third_party/mbedtls/repo/library/entropy_poll.c         \
+    third_party/mbedtls/repo/library/ssl_cookie.c           \
+    third_party/mbedtls/repo/library/ssl_ciphersuites.c     \
+    third_party/mbedtls/repo/library/ssl_cli.c              \
+    third_party/mbedtls/repo/library/ssl_srv.c              \
+    third_party/mbedtls/repo/library/ssl_ticket.c           \
+    third_party/mbedtls/repo/library/ssl_tls.c              \
+    third_party/mbedtls/repo/library/aes.c                  \
+    third_party/mbedtls/repo/library/ecp.c                  \
+    $(NULL)
+
+include $(BUILD_STATIC_LIBRARY)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE:= ot-cli
+LOCAL_MODULE_TAGS := eng
+
+LOCAL_C_INCLUDES                                         := \
+    $(LOCAL_PATH)/include                                   \
+    $(LOCAL_PATH)/src                                       \
+    $(LOCAL_PATH)/src/cli                                   \
+    $(LOCAL_PATH)/src/core                                  \
+    $(LOCAL_PATH)/src/posix/platform                        \
+    $(LOCAL_PATH)/third_party/mbedtls/repo/include          \
+    $(NULL)
+
+LOCAL_CFLAGS                                                                += \
+    -D_GNU_SOURCE                                                              \
+    -DSPINEL_PLATFORM_HEADER=\"spinel_platform.h\"                             \
+    -DOPENTHREAD_CONFIG_FILE=\<openthread-config-android.h\>                   \
+    -DOPENTHREAD_FTD=1                                                         \
+    -DOPENTHREAD_POSIX=1                                                       \
+    -DOPENTHREAD_POSIX_APP=2                                                   \
```

Comment:
`OPENTHREAD_POSIX_APP=1`?

Reply:
2 means the app is to be compiled as cli mode. and 1 means the app is to be compiled as spinel/ncp mode.